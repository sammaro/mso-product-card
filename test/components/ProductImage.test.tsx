import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('should display the product without image', () => {
    const wrapper = renderer.create(<ProductImage img="http://img.ar" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should display the product with image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
