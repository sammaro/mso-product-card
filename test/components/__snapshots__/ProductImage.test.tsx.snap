// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should display the product with image 1`] = `
<div
  className="productCard undefined"
>
  <img
    className="productImg undefined"
    src="./cofee-mug.png"
  />
</div>
`;

exports[`ProductImage should display the product without image 1`] = `
<img
  className="productImg undefined"
  src="http://img.ar"
/>
`;
